Extension/src/LanguageServer/client.ts

```diff
@@ -756,7 +756,7 @@ class DefaultClient implements Client {
             return;
         }
 
-        let showCompileCommandsSelection: PersistentState<boolean> = new PersistentState<boolean>("CPP.showCompileCommandsSelection", true);
+        let showCompileCommandsSelection: PersistentFolderState<boolean> = new PersistentFolderState<boolean>("CPP.showCompileCommandsSelection", true, this.Name);
```

Comment:
I believe the 3rd parameter is supposed to be an absolute path.  e.g. `this.RootPath`